### PR TITLE
Make a simple "var" inferrer.

### LINKED ISSUE #1408: Infer "var" type

### PR DESCRIPTION
**Pull request by matozoid**
Fixes #1408.

@ftomassetti - what is the correct way to deal with errors? What if we try to resolve `var` in a position where it was not allowed to be? What if there is no initializer to take the type from?

@ftomassetti - is the test written well?

### CODE REVIEW

File: javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/JavaParserFacade.java [hunk 1]
```diff
@@ -473,6 +473,9 @@ protected ResolvedType convertToUsage(com.github.javaparser.ast.type.Type type,
         } else if (type instanceof UnionType) {
             UnionType unionType = (UnionType) type;
             return new ResolvedUnionType(unionType.getElements().stream().map(el -> convertToUsage(el, context)).collect(Collectors.toList()));
+        } else if (type instanceof VarType) {
+            final VariableDeclarator variableDeclarator = (VariableDeclarator)type.getParentNode().get();
+            return variableDeclarator.getInitializer().get().calculateResolvedType();
```
Comment (ftomassetti) [inline javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/JavaParserFacade.java]: This is what I would do

**Review by ftomassetti — APPROVED** [review]
Wow, this seems surprisingly simple!

File: javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/javaparser/VarTypeTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            .setSymbolResolver(new JavaSymbolSolver(typeSolver)));
+
+    @Test
+    public void resolveAPrimitive() {
+        CompilationUnit ast = javaParser.parse(ParseStart.COMPILATION_UNIT, provider("class X{var abc = 1;}")).getResult().get();
+        VarType varType = ast.findFirst(VarType.class).get();
+
+        ResolvedType resolvedType = varType.resolve();
+
+        assertEquals(ResolvedPrimitiveType.byName("int"), resolvedType);
```
Comment (ftomassetti) [inline javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/resolution/javaparser/VarTypeTest.java]: I would use ResolvedPrimitiveType.INT

**Comment by ftomassetti** [comment]
Tests look good.

Regarding the fact an initializer is not present: I would throw an IllegalStateException, but we could create also some hierarchy of exceptions to represent these semantic problems

**Comment by matozoid** [comment]
@ftomassetti I've added the error handling. In my experience exception hierarchies don't really add much - people just want to know what went wrong and fix it, so a helpful message is better :-)

**Comment by matozoid** [comment]
Oh, @ftomassetti - in which package should this test be?

**Comment by ftomassetti** [comment]
I think under resolution is the best option. We have a typeinference package but it is about something different

**Comment by matozoid** [comment]
Okay, so I guess this is good to go!

**Pull request merged**